models/articles.models.js

```diff
@@ -1,14 +1,46 @@
-const db = require('../db/connection')
+const db = require("../db/connection");
 
 exports.retrieveArticleById = (article_id) => {
-  return db.query(`
+  return db
+    .query(
+      `
     SELECT * FROM articles 
     WHERE article_id = $1
-  `, [article_id])
-  .then(({rows}) => {
-    if(rows.length === 0) {
-      return Promise.reject({msg: 'article does not exist'})
-    }
-    return rows[0]
-  })
-}
\ No newline at end of file
+  `,
+      [article_id]
+    )
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ msg: "article does not exist" });
+      }
+      return rows[0];
+    });
+};
+
+exports.retrieveAllArticles = () => {
+  const promiseOne = db.query(
+    "SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.votes, articles.article_img_url FROM articles ORDER BY created_at DESC"
+  );
+  const promiseTwo = db.query(`
+  SELECT COUNT(article_id), article_id FROM comments GROUP BY article_id
+  `);
+
+  return Promise.all([promiseOne, promiseTwo]).then((result) => {
+    const articles = result[0].rows;
+    const comments = result[1].rows;
+
```

Comment:
Nice solution but you can combine the two queries into one, if you JOIN the comments table in the first query you would then have access to comments.comment_id - which you could then use in the COUNT function. You can use AS to name a column for these counts, so you can create a comment_count column in the query instead of adding them to the object below. 

If you do that, you can do it only having one query to the database and won't need a Promise.all - or need to manipulate the object afterwards.